### PR TITLE
Bump managed kotlin version to 1.7.10

### PR DESCRIPTION
**Pull request by wetted**
satisfies https://github.com/micronaut-projects/micronaut-kotlin/issues/414

### CODE REVIEW

**Review by sdelamo — CHANGES_REQUESTED** [review]
Misses docs

**Comment by timyates** [comment]
We see lots of 1.6.21 in the dependency list

https://ge.micronaut.io/s/bjjijqfihhmsu/dependencies?dependencies=1.6.21&expandAll

But I believe these are due to the POM for [org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:1.6.4](https://search.maven.org/artifact/org.jetbrains.kotlinx/kotlinx-coroutines-jdk8/1.6.4/jar) having a dependency on it

I think none of these are being pulled by Gradle, but when they are it's picking the correct one

https://ge.micronaut.io/s/bjjijqfihhmsu/dependencies?dependencies=1.6.21&expandAll&focusedDependency=WzQxLDEsNDI0OCxbNiwwLFsyNl1dXQ

@melix is my understanding correct?

File: buildSrc/src/main/groovy/io/micronaut/build/internal/ext/DefaultMicronautCoreExtension.java [hunk 1]
```diff
@@ -60,6 +60,11 @@ public void usesMicronautTestKotest() {
         addTestImplementationDependency("kotest");
     }
 
+    @Override
```
Comment (sdelamo) [inline buildSrc/src/main/groovy/io/micronaut/build/internal/ext/DefaultMicronautCoreExtension.java]: can we get rid of kotest4 ?

File: buildSrc/src/main/groovy/io/micronaut/build/internal/ext/MicronautCoreExtension.java [hunk 2]
```diff
@@ -28,4 +28,6 @@ public interface MicronautCoreExtension {
 
     void usesMicronautTestKotest();
 
+    void usesMicronautTestKotest5();
```
Comment (sdelamo) [inline buildSrc/src/main/groovy/io/micronaut/build/internal/ext/MicronautCoreExtension.java]: can we just use kotest5 instaed of mantaining both?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/groovy/io/micronaut/build/internal/ext/DefaultMicronautCoreExtension.java (hunk 1 above)
Comment (timyates) [reply buildSrc/src/main/groovy/io/micronaut/build/internal/ext/DefaultMicronautCoreExtension.java]: Gone

File: buildSrc/src/main/groovy/io/micronaut/build/internal/ext/MicronautCoreExtension.java (hunk 2 above)
Comment (timyates) [reply buildSrc/src/main/groovy/io/micronaut/build/internal/ext/MicronautCoreExtension.java]: Gone

Comment (sdelamo) [reply buildSrc/src/main/groovy/io/micronaut/build/internal/ext/MicronautCoreExtension.java]: why create a new method instead of keeping old one?

Comment (timyates) [reply buildSrc/src/main/groovy/io/micronaut/build/internal/ext/MicronautCoreExtension.java]: Fixed in https://github.com/micronaut-projects/micronaut-core/pull/8074/commits/624810d8a859c728f53593a814bd9a3a5247d0e2

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
ping @melix

**Review by melix — APPROVED** [review]

**Pull request merged**